nhsuk/nhsuk-service-manual
# Add gitleaks pre-commit

## Description
Add gitleaks as a pre-commit so that secret scanning is performed before a commit is allowed to go ahead.
Also adds gitleaks automatically if running in Gitpod.

TO TEST:

1. Open PR in Gitpod or open branch locally (if running locally you will need to follow gitleaks installation instructions in .husky/gitleaks/config/README-gitleaks.md).
2. Add the following line to app/scripts/main.js **aws_secret="AKIAIMNOJVGFDXXXE4OA"**
3. Try and commit that file and gitleaks should detect it as a secret and cancel the commit.

### Related issue
https://nhsd-jira.digital.nhs.uk/browse/TM-2720 

## Checklist


- [ ] Tested against the [NHS.UK frontend testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [X] Code follows the [NHS.UK frontend coding standards](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/coding-standards.md)
- [X] Version number has been increased in `package.json` (using [SEMVER](https://semver.org/))
- [ ] [Whats new page](https://service-manual.nhs.uk/whats-new) entry
- [ ] Page updated date
